File: src/main/java/ru/atsutsiev/sd/refactoring/servlet/GetProductsServlet.java
package ru.atsutsiev.sd.refactoring.servlet;

import ru.atsutsiev.sd.refactoring.database.ProductsTable;

import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.PrintWriter;
import java.sql.SQLException;

/**
 * @author atsutsiev
 */
public class GetProductsServlet extends HttpServlet {
    private final ProductsTable database;

    public GetProductsServlet(final ProductsTable database) { this.database = database; }

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
        try (PrintWriter writer = response.getWriter()) {
            writer.println(database.queryAsHTML("all"));
        } catch (final SQLException e) {
            throw new RuntimeException("SQLException: " + e); // I'm sorry for this
        }
        response.setContentType("text/html");
        response.setStatus(HttpServletResponse.SC_OK);
    }
}

File: src/main/kotlin/ru/atsutsiev/sd/refactoring/servlet/GetProductsServlet.kt
package ru.atsutsiev.sd.refactoring.servlet

import ru.atsutsiev.sd.refactoring.database.models.Products
import java.io.IOException
import java.sql.SQLException
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * @author atsutsiev
 */
class GetProductsServlet(private val model: Products) : HttpServlet() {
    @Throws(IOException::class)
    override fun doGet(request: HttpServletRequest?, response: HttpServletResponse) {
        try {
            response.writer.apply { print(model.queryAsHTML("all")) }
        } catch (e: SQLException) {
            throw RuntimeException("SQLException: $e") // I'm sorry for this
        }

        response.apply {
            contentType = "text/html"
            status = HttpServletResponse.SC_OK
        }
    }
}